main.c

```diff
@@ -0,0 +1,572 @@
+                
+                p->next = newStudent;
+            }
+        } else 
+            printf("Invalid level or class: %d, %d\n", level, class);
+ 
+    }
+
+}
+
+// Function to print the details of each student in the school array
+void printAllStudents() {
+    for (int level = 0; level < LEVELS; level++) {
+        for (int class = 0; class < CLASS; class++) {
+            printf("Level: %d   ", level + 1); // Adjust level and class to one-based indexing
+            printf("Class: %d\n\n", class + 1); // Adjust level and class to one-based indexing
+            Student* p = school[level][class];
+            while (p != NULL) {
+                printf("Name: %s %s\n", p->firstName, p->lastName);
+                printf("Phone Number: %s\n", p->phoneNumber);
+                printf("Grades: ");
+                for (int i = 0; i < 10; i++) {
+                    printf("%d ", p->grades[i]);
+                }
+                printf("\n\n");
+                p = p->next;
+            }
+        }
+    }
+}
+
+// Function to free the memory allocated for each student
+void freeSchoolMemory(Student* school[LEVELS][CLASS]) {
+    for (int level = 0; level < LEVELS; level++) {
+        for (int class = 0; class < CLASS; class++) {
+            Student* p = school[level][class];
+            while (p != NULL) {
+                Student* temp = p;
+                p = p->next;
+                free(temp);
+            }
+        }
+    }
+}
+
+void deleteStudent() {
+    char phoneNumber[MAXNUMP];
+    printf("Enter the phone number of the student to delete: ");
+    scanf("%s", phoneNumber);
+    for (int level = 0; level < LEVELS; level++) {
+        for (int class = 0; class < CLASS; class++) {
+            Student* p = school[level][class];
+            Student* prev = NULL;
+
+            while (p != NULL) {
+                if (strcmp(p->phoneNumber, phoneNumber) == 0) {
+                    // Student with matching phone number found, delete it
+                    if (prev == NULL) {
+                        // It's the first node, update school array
+                        school[level][class] = p->next;
+                    } else {
+                        // Update previous node's next pointer
+                        prev->next = p->next;
+                    }
+                    free(p);
+                    printf("Student with phone number %s deleted successfully.\n", phoneNumber);
+                    return; 
+                }
+                prev = p;
+                p = p->next;
+            }
+        }
+    }
+
+    // If the function reaches this point, the student was not found
+    printf("Student with phone number %s not found.\n", phoneNumber);
+}
+
+void insertNewStudent() {
+    char firstName[50];
+    char lastName[50];
+    char phoneNumber[11];
+    int level, class;
+
+    
+    
+    printf("Enter the student's first name: ");
+    get_input(firstName,MAXNUMF);
```

Comment:
What if get_input fails? It seems like you didn't fully understand the function
before using it.